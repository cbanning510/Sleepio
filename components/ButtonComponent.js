import React from 'react';
import {StyleSheet, TouchableOpacity, Text} from 'react-native';

const ButtonComponent = ({onPress, title}) => {
  return (
    <TouchableOpacity style={styles.buttonContainer} onPress={onPress}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  buttonText: {
    color: 'white',
  },
  buttonContainer: {
    margin: 14,
    padding: 6,
    paddingHorizontal: 10,
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 4,
  },
});

export default ButtonComponent;
