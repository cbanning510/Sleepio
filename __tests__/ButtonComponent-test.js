import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import ButtonComponent from '../components/ButtonComponent';

const title = 'Title';

it('renders correctly', () => {
  const tree = renderer.create(<ButtonComponent title={title} />);
  expect(tree).toMatchSnapshot();
});

it('displays correct title text', () => {
  const {getAllByText} = render(<ButtonComponent title={title} />);
  expect(getAllByText('Title').length).toBe(1);
});
