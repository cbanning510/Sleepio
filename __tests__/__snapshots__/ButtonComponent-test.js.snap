// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "white",
      "borderRadius": 4,
      "borderWidth": 1,
      "margin": 14,
      "opacity": 1,
      "padding": 6,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    Title
  </Text>
</View>
`;
